[GTK] Move ValidationBubble implementation from WebCore to WebKit

carlosgcampos opened the pull request:
#### 6c551feae68d2196e3e3b7bda2d1358ee125bc17
<pre>
[GTK] Move ValidationBubble implementation from WebCore to WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=300618">https://bugs.webkit.org/show_bug.cgi?id=300618</a>

Reviewed by Michael Catanzaro.

It&apos;s only used by WebKit in the UI process, so we don&apos;t need it in
WebCore. We can&apos;t move the base class to WebKit yet because it&apos;s also
used by WebKitLegacy, so for now this patch makes the base class
asbtract for GTK port and uses a derived class from WebKit.

Canonical link: <a href="https://commits.webkit.org/301538@main">https://commits.webkit.org/301538@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d8feb718bf21815f8f92bc1ce3eb1ada488cfe52

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126288 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45942 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36773 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133059 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/78045 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128159 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46604 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54485 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/96146 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/78045 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129236 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37283 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112958 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/76624 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/36183 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31138 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76525 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107067 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31373 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135756 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53028 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40739 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/104646 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53492 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109201 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/104344 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49791 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28124 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50408 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19748 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52930 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58757 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52242 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55586 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53961 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebCore/platform/ValidationBubble.h:
```diff
@@ -62,41 +60,36 @@ class ValidationBubble : public RefCounted<ValidationBubble> {
-    {
-        return adoptRef(*new ValidationBubble(view, WTFMove(message), settings, WTFMove(callback)));
-    }
+    virtual ~ValidationBubble() = default;
```
Let's add a warning comment to reduce risk of future error:

```suggestion
    // ValidationBubbleGtk is inherited from ValidationBubble.
    virtual ~ValidationBubble() = default;
```

mcatanzaro reviewed (approved):
Well, honestly, this is a mess, but it works. I'm going to click the Approve button so you can land this if you don't want to spend more time on it. I can't think of anything better that does not require a bunch more work.

The _nice_ way to do this would be to have just a delegate interface in WebCore, and require WebKitLegacy/WebKit layers to provide the actual delegate implementation. And in fact, we already have WebValidationMessageClient; we might be able to use that without needing a separate delegate for ValidationBubble. But this would be significantly more work, so I'm not going to request it. And you'd really need macOS and XCode to make those changes; that's too much to try doing from Linux.

carlosgcampos committed:
[GTK] Move ValidationBubble implementation from WebCore to WebKit
https://bugs.webkit.org/show_bug.cgi?id=300618

Reviewed by Michael Catanzaro.

It's only used by WebKit in the UI process, so we don't need it in
WebCore. We can't move the base class to WebKit yet because it's also
used by WebKitLegacy, so for now this patch makes the base class
asbtract for GTK port and uses a derived class from WebKit.

Canonical link: https://commits.webkit.org/301538@main
